src/vsphere_cpi/spec/integration/vm_host_rule_spec.rb

```diff
@@ -0,0 +1,402 @@
+      @second_cluster_name = fetch_and_verify_cluster('BOSH_VSPHERE_CPI_SECOND_CLUSTER')
+      @datastore_shared_pattern = fetch_property('BOSH_VSPHERE_CPI_SHARED_DATASTORE')
+      @first_cluster_datastore=fetch_property('BOSH_VSPHERE_CPI_DATASTORE_PATTERN')
+      @first_host_group = fetch_property('BOSH_VSPHERE_CPI_FIRST_CLUSTER_FIRST_HOST_GROUP')
+      @second_host_group = fetch_property('BOSH_VSPHERE_CPI_FIRST_CLUSTER_SECOND_HOST_GROUP')
+      @third_host_group = fetch_property('BOSH_VSPHERE_CPI_SECOND_CLUSTER_FIRST_HOST_GROUP')
+      @second_cluster_resource_pool_name = fetch_and_verify_resource_pool('BOSH_VSPHERE_CPI_SECOND_CLUSTER_RESOURCE_POOL', @second_cluster_name)
+    end
+
+    let(:options) do
+      cpi_options(
+        datacenters: [{
+          clusters: [
+            {
+              @cluster_name => {
+                host_group:  @first_host_group,
+              }
+            }
+          ]
+        }]
+      )
+    end
+    let(:cpi) do
+      VSphereCloud::Cloud.new(options)
+    end
+    let(:vm_type) do
+      {
+        'ram' => 512,
+        'disk' => 2048,
+        'cpu' => 1,
+      }
+    end
+    let(:cluster_mob) do
+      cpi.client.cloud_searcher.get_managed_object(
+        VimSdk::Vim::ClusterComputeResource, name: @cluster_name)
+    end
+    let(:second_cluster_mob) do
+      cpi.client.cloud_searcher.get_managed_object(
+        VimSdk::Vim::ClusterComputeResource, name: @second_cluster_name)
+    end
+    let(:first_host_group_mob) do
+      cluster_mob.configuration_ex.group.find do |group|
+        group.name == @first_host_group && \
+            group.is_a?(VimSdk::Vim::Cluster::HostGroup)
+      end
+    end
+    let(:second_host_group_mob) do
+      cluster_mob.configuration_ex.group.find do |group|
+        group.name == @second_host_group && \
+            group.is_a?(VimSdk::Vim::Cluster::HostGroup)
+      end
+    end
+    let(:third_host_group_mob) do
+      second_cluster_mob.configuration_ex.group.find do |group|
+        group.name == @third_host_group && \
+            group.is_a?(VimSdk::Vim::Cluster::HostGroup)
+      end
+    end
+    let(:hosts_in_first_host_group) { first_host_group_mob.host.map(&:name) }
+    let(:hosts_in_second_host_group) { second_host_group_mob.host.map(&:name) }
+    let(:hosts_in_third_host_group) { third_host_group_mob.host.map(&:name) }
+
+
+    context 'when there is only one vSphere cluster in an AZ' do
+      let(:hosts_in_first_host_group) { first_host_group_mob.host.map(&:name) }
+      let(:hosts_in_second_host_group) { second_host_group_mob.host.map(&:name) }
+
+      context 'when one host_group is specified in the cluster' do
+        it 'should create one VM on one of the hosts in the host groups' do
+          simple_vm_lifecycle(cpi, '', vm_type, get_network_spec) do |vm_id|
+            vm = cpi.vm_provider.find(vm_id)
+            expect(hosts_in_first_host_group).to include(vm.mob.runtime.host.name)
```

Comment:
you should also test as part of this integration test that VM is added to the VM group and the rule is present. This spec is no where testing anything about VM Host Rule creation except that the count is 0 at the end

Reply:
On it.

Reply:
Have created a follow up story to resolve this.